neo/rawio/intanbinaryrawio.py

```diff
@@ -0,0 +1,394 @@
+"""
+Support for Intan's binary file mode which is turned on by setting the save
+option in Intan to 'One File Per Signal Type' or 'One File Per Channel'.
+The reader for this format is for rhd only since it uses the rhd header.
+Need to confirm if rhs can also use this save format.
+
+RHD supported 1.x, 2.x, and 3.0, 3.1, 3.2
+See:
+  * http://intantech.com/files/Intan_RHD2000_data_file_formats.pdf
+  * http://intantech.com/files/Intan_RHS2000_data_file_formats.pdf
+
+Author: Zach McKenzie, based on Samuel Garcia's IntanRawIO
+"""
+from pathlib import Path
+
+import numpy as np
+from collections import OrderedDict
+from packaging.version import Version as V
+
+from .baserawio import (BaseRawIO, _signal_channel_dtype, _signal_stream_dtype,
+                _spike_channel_dtype, _event_channel_dtype, _common_sig_characteristics)
+
+from .intanrawio import (read_variable_header, rhd_global_header_base, rhd_global_header_part1,
+                         rhd_global_header_v11, rhd_global_header_v13, rhd_global_header_v20,
+                         rhd_global_header_final, rhd_signal_group_header,
+                         rhd_signal_channel_header, stream_type_to_name)
+
+
+class IntanBinaryRawIO(BaseRawIO):
+    """
+    Class for processing Intan Data when saved in binary format. Requires an
+    `info.rhd` as well one file per signal stream or one file per channel
+
+    Parameters
+    ----------
+    dirname: str, Path
+        The root directory containing the info.rhd file
+
+    """
+    extensions = ['rhd', 'dat']
+    rawmode = 'one-dir'
+
+    def __init__(self, dirname=''):
+        BaseRawIO.__init__(self)
+        self.dirname = dirname
+
+    def _source_name(self):
+        return self.dirname
+
+    def _parse_header(self):
+
+        dir_path = Path(self.dirname)
+        assert dir_path.is_dir(), ('IntanBinaryRawIO requires the root '
+                                   'directory containing info.rhd')
+
+        header_file = dir_path / 'info.rhd'
+
+        for file in possible_raw_files:
+            if (dir_path / file).is_file():
+                stream_mode = True
```

Comment:
`stream_mode` is used to differentiate between one signal per file and one channel per file, right? I think it would be better to use intan relevant terminology. I suggest using "one_signal_per_file` and `one_channel_per_file` as the flags that alter the behavior. Even more so because you then seem to use `stream_index` for enumerating the signals so there is some overloading of stream in this class.

In general,  I think it is better to use domain relevant varaible names instead of variable names that allude to the data structures utilized on the code.

Reply:
I think that is a great point. I realized I would need to distinguish later in the writing (I originally only wrote for one_signal_per_file) and so just tossed that in. But you're write. Better names will make reading easier in the future!  I will push a fix for that soon.

Reply:
Done. :)

And I changed my utility functions to better describe their actions.

Reply:
Cool. I will take another look.

Also I just talked with Sam and he is fine with the data. I will push it to gin this week.

Reply:
Thanks @h-mayorquin! I need to sign up for a gin account but have just been too lazy...